chain/events/tscache.go

```diff
@@ -95,14 +104,18 @@ func (tsc *tipSetCache) getNonNull(height abi.ChainEpoch) (*types.TipSet, error)
 }
 
 func (tsc *tipSetCache) get(height abi.ChainEpoch) (*types.TipSet, error) {
```

Comment:
This is a bit nasty but I'm trying to avoid calling into the storage subsystem while holding a lock.